### PR TITLE
docs(template-syntax/structural-directives): refresh both guides

### LINKED ISSUE #2303: [2.0] Drop `<template [ngIf]="">` syntax from docs & examples
While working on the `NgIf` API docs we first thought about documenting:
- `<single-root *ngIf=exp>`
- `<template [ngIf]=exp><root1><root2></template>

However this is confusing, we will drop the `<template []>` notation from the API docs - note that it will still be supported and there is no plan to remove the support.

The preferred way will be `<ng-container *ngIf=exp><root1><root2></ng-container>`

The `<ng-container>` tag is used to group nodes together and is not stamped out in the resulting DOM. It is not a directive but a special angular construct.

We think that this change will remove some confusion and hard to remember syntax.

Please ping me if you need additional info.

/cc @IgorMinar

### PR DESCRIPTION
**Pull request by wardbell**
Refresh both guides for style, accuracy, understanding
Move details of structural directives from template-syntax to structural-directives guide
Add `<ng-container>` to structural-directives
Fix samples in both guides
Touch up glossary
Better conformance to google doc guidelines: we->you
closes #2303, #2885

Todo
* [x] Update template-syntax e2e tests [too few tests; none touched; revisit later]
* [x] Update structural-directives e2e tests

### CODE REVIEW

File: public/docs/ts/latest/guide/template-syntax.jade [hunk 1]
```diff
@@ -513,26 +569,30 @@ table
...
+  As mentioned previously, evaluation of a template expression should have no visible side effects. 
+  The expression language itself does its part to keep you safe. 
+  You can't assign a value to anything in a property binding expression nor use the increment and decrement operators.
 
-  Of course, our expression might invoke a property or method that has side effects. Angular has no way of knowing that or stopping us.
+  Of course, the expression might invoke a property or method that has side effects. Angular has no way of knowing that or stopping you.
 
-  The expression could call something like `getFoo()`. Only we know what `getFoo()` does.
-  If `getFoo()` changes something and we happen to be binding to that something, we risk an unpleasant experience. Angular may or may not display the changed value. Angular may detect the change and throw a warning error. Our general advice: stick to data properties and to methods that return values and do no more.
+  The expression could call something like `getFoo()`. Only directive know what `getFoo()` does.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/template-syntax.jade]: Possible missing word? Maybe "the"?

"Only directive know what `getFoo()` does."

File: public/docs/ts/latest/guide/template-syntax.jade [hunk 2]
```diff
@@ -541,7 +601,7 @@ block dart-type-exceptions
 :marked
   ### Remember the brackets
   The brackets tell Angular to evaluate the template expression.
-  If we forget the brackets, Angular treats the string as a constant and *initializes the target property* with that string.
+  If directive forget the brackets, Angular treats the string as a constant and *initializes the target property* with that string.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/template-syntax.jade]: "If the directive doesn't have the brackets" or something similar that fixes "directive forget"?

Comment (wardbell) [inline public/docs/ts/latest/guide/template-syntax.jade]: How about "if you omit the brackets"

I thought I fixed all the "we" to "you". Hmmm

Comment (kapunahelewong) [inline public/docs/ts/latest/guide/template-syntax.jade]: I think that was the first of a rogue find and replace that sprinkled "directive" in places "you" or "we" seems to have been. I think I changed all the rest of them. I just went back and changed this one and pushed up.

File: public/docs/ts/latest/guide/template-syntax.jade (hunk 1 above)
Comment (wardbell) [inline public/docs/ts/latest/guide/template-syntax.jade]: Oh my!  Some kind of search and replace changed "your" or "we" to "directive".  Seems to have been limited to this location. Fixed it.

**Pull request merged**